substrate/frame/identity/src/lib.rs

```diff
@@ -331,36 +448,34 @@ pub mod pallet {
 			info: Box<T::IdentityInformation>,
 		) -> DispatchResultWithPostInfo {
 			let sender = ensure_signed(origin)?;
-			let encoded_byte_size = info.encoded_size() as u32;
-			let byte_deposit =
-				T::ByteDeposit::get().saturating_mul(<BalanceOf<T>>::from(encoded_byte_size));
-
-			let mut id = match <IdentityOf<T>>::get(&sender) {
-				Some(mut id) => {
-					// Only keep non-positive judgements.
-					id.judgements.retain(|j| j.1.is_sticky());
-					id.info = *info;
-					id
-				},
-				None => Registration {
-					info: *info,
-					judgements: BoundedVec::default(),
-					deposit: Zero::zero(),
-				},
+
+			let (mut id, username) = match <IdentityOf<T>>::get(&sender) {
+				Some((mut id, maybe_username)) => (
+					{
+						// Only keep non-positive judgements.
+						id.judgements.retain(|j| j.1.is_sticky());
+						id.info = *info;
+						id
+					},
+					maybe_username,
+				),
+				None => (
+					Registration {
+						info: *info,
+						judgements: BoundedVec::default(),
+						deposit: Zero::zero(),
+					},
+					None,
+				),
 			};
 
+			let new_deposit = Self::calculate_identity_deposit(&id.info);
```

Comment:
Seems like a good candidate for using the `Consideration` API.

Reply:
I think there is another effort to migrate this whole pallet. Didn't want to mix that into this.